Docs: Jupyter Notebooks

seeM opened the pull request:
Addresses https://github.com/posit-dev/positron/issues/7545.

It does feel short, but I'm also not too sure what else to add at this stage, since most things are either covered in VS Code docs or Positron docs about the various panes. Happy to expand based on the initial feedback.

Full disclosure: I used an LLM to write a first draft but it wasn't very good so I edited ~90% of it from there.

isabelizimm commented on jupyter-notebooks.qmd:
```diff
@@ -0,0 +1,33 @@
+[Jupyter Notebooks](https://jupyter-notebook.readthedocs.io/en/latest/notebook.html) offer a flexible, interactive UI for combining code, prose, and visualizations.
+In Positron, Jupyter Notebooks are fully supported and integrated into a complete data science IDE.
+
+## Why use Jupyter Notebooks in Positron over other IDEs?
```
I think this section could be under the `## Using Jupyter Notebooks in Positron` and call it `### Jupyter Notebook integrations` or similar

seeM replied:
Added to `accept.txt`, thanks!

isabelizimm reviewed (commented):
This is a great start! I think it's alright that it is a little shorter, but a few other possibilities

- Any setting/commands people should be aware of
- Plots pane integration
- Small section on Variables pane integration

austin3dickey commented on jupyter-notebooks.qmd:
```diff
@@ -0,0 +1,33 @@
+### Prerequisites
+
+Positron comes prepackaged with Jupyter kernels for R and Python.
+Once you've [configured an R or Python environment](start.qmd) for Positron, you do not need to install any additional dependencies into your environment before using notebooks.
```
The link to `start.qmd` feels a little confusing; maybe either get rid of it or link to `managing-interpreters.qmd` instead?

austin3dickey commented on jupyter-notebooks.qmd:
```diff
@@ -0,0 +1,33 @@
+
+When you first open a Jupyter Notebook in Positron, it automatically selects the most appropriate interpreter based on the notebook's language and the current workspace configuration.
+
+You can manually select a different interpreter for the notebook using the kernel selector in the notebook toolbar.
```
And `r-installations.qmd` :)

austin3dickey reviewed (commented):
Looks great; agreed with Isabel on the possible additions.

seeM commented on jupyter-notebooks.qmd:
```diff
@@ -0,0 +1,33 @@
+
+1. Notebooks work out of the box.
+   You don't need to install any additional dependencies into your R or Python environments.
+1. The Variables Pane provides an expandable overview of the variables in your notebook, automatically updating as you work.
```
Using `1.` for Markdown lists is a trick I've gotten used to. It still renders with increasing numbers but you don't have to change the numbers in the source file when you rearrange the list.

I agree bullet points are fine here though!

seeM commented:
Thanks for the reviews! I've addressed most points except:

> - Any setting/commands people should be aware of

I'm not sure of any Positron-specific settings or commands at this point. Are there any that come to mind?

> - Plots pane integration

There is currently no integration, also hoping to move toward this soon. What did you have in mind?

> - Small section on Variables pane integration

I'm not sure what else to add in addition to the bullet point at the top, and don't want to overlap with the Variables Pane docs when we add them.

isabelizimm reviewed (approved):
> There is currently no integration, also hoping to move toward this soon. What did you have in mind?

I was thinking maybe mentioning that plots are inline in notebooks (vs scripts that use the plots pane), but it's probably fair that people will notice that from a screen shot or trying it out.

Ah, yeah, we can fill out more info in the Variables pane doc when that comes in!

This looks good to me 👾

juliasilge reviewed (approved):
Thank you so much! 🙌 Feel free to merge after addressing these style nits.

seeM commented on jupyter-notebooks.qmd:
```diff
@@ -0,0 +1,35 @@
+title: "Jupyter Notebooks"
+---
+
+[Jupyter Notebooks](https://jupyter-notebook.readthedocs.io/en/latest/notebook.html) offer a flexible, interactive UI for combining code, prose, and visualizations. In Positron, Jupyter Notebooks are fully supported and integrated into a complete data science IDE.
```
My intended point was that in Positron, Jupyter Notebooks should feel like they fit well into a feature-filled IDE, which isn't what standard Jupyter Notebook users are used to. I also suspect most readers will only view a few pages and not read end-to-end.

seeM replied:
I removed this sentence since I think the intended point comes across better in the bullet points below.

seeM commented on jupyter-notebooks.qmd:
```diff
@@ -0,0 +1,35 @@
+
+### Selecting an interpreter
+
+When you first open a Jupyter Notebook, Positron automatically selects the most appropriate interpreter based on the notebook's language and the current workspace.
```
Good point! How about "Positron automatically selects an interpreter based on..."? I suspect we'll be changing these heuristics in our current focus so don't want to commit just yet.

juliasilge commented on jupyter-notebooks.qmd:
```diff
@@ -0,0 +1,35 @@
+
+Positron comes prepackaged with Jupyter kernels for R and Python. Once you've [configured an R or Python environment](managing-interpreters.qmd) for Positron, you do not need to install any additional dependencies into your environment before using notebooks.
+
+If an environment installed on your computer isn't available in Positron, you may want to read more about how Positron [discovers R](r-installations.qmd) and [Python installations](python-installations.qmd).
```
HA yes, I have done alternating on some other projects that support both Python and R. 😅
